src/WebJobs.Script/Host/ScriptHostIdProvider.cs

```diff
@@ -33,6 +34,7 @@ public Task<string> GetHostIdAsync(CancellationToken cancellationToken)
             string hostId = _config[ConfigurationSectionNames.HostIdPath];
             if (hostId == null)
             {
+                // if the user hasn't configured an explicit ID, we generate the default ID.
```

Comment:
Ah this answers the question I had earlier -- so if a user doesn't configure explicit ID, then we'd return a new hash each time? 

If the uniqueSlotName is doesn't change during the lifecycle of the ScriptHostIdProvider, and hashes to the same value each time, I'd consider storing the hash result in an attribute as an optimization

Reply:
For a given site name, the host ID we return here will of course remain constant. Putting caching in place is beyond the scope of this PR, and we'd have to be careful. This ScriptHostIDProvider instance lives across host instances so we have to be careful with caching. From a layering point of view, it's always possible for upstream consumers of this API to cache the value if they want to.